torch/onnx/symbolic_opset9.py

```diff
@@ -725,16 +729,43 @@ def prelu(g, self, weight):
 def silu(g, input):
     return g.op("Mul", input, g.op("Sigmoid", input))
 
+
 def mish(g, input):
     return g.op("Mul", input, g.op("Tanh", g.op("Softplus", input)))
 
+
+def _compatible_float_cast(g, op_name, self, *args, other_operands=None, opset_before=None, **kwargs):
```

Comment:
Many operators, such as `Max-13` allow `T : tensor(float16), tensor(float), tensor(double), tensor(bfloat16)` as input, but the cast-op-cast pattern implemented here always use `Float`.

Is there anything we can do to handle mixed precision too besides float32?

Reply:
Maybe using `torch.is_autocast_enabled()` and checking whether the op do accept fp16 or bfloat16

Reply:
Good point. So I also added another attribute "target_float_t" as the prospective float types in casting.

> Maybe using torch.is_autocast_enabled() and checking whether the op do accept fp16 or bfloat16

The API `sym_help._is_fp(self)` can already tell if it can support all kinds of float types.